source/dub/packagemanager.d

```diff
@@ -533,7 +533,18 @@ class PackageManager {
 				try foreach( pdir; iterateDirectory(path) ){
 					logDebug("iterating dir %s entry %s", path.toNativeString(), pdir.name);
 					if( !pdir.isDirectory ) continue;
-					auto pack_path = path ~ (pdir.name ~ "/");
+
+					// Search for a single directory within this directory which happen to be a prefix of pdir
+					// This is to support new folder structure installed over the ancient one.
+					FileInfo subdir;
+					foreach( pdir2; iterateDirectory(path ~ (pdir.name ~ "/")) )
+					{
+						if (pdir.name.startsWith(pdir2.name)) // eg: package vibe-d will be in "vibe-d-x.y.z/vibe-d"
+							subdir = pdir2;
+					}
+					if( !subdir.isDirectory ) continue;
```

Comment:
Actually the former layout isn't supported at all anymore. 

It won't get list by the scan, so the new layout will be installed on top of it (which is messy, but works in almost all cases: the problem case being when there was already a subdir called package-name with dub.json itself. Which I think is incredibly rare).